### PR TITLE
[RDY] Clear prompt_for_number messages

### PR DESCRIPTION
**Pull request by glacambre**

### CODE REVIEW

File: src/nvim/misc1.c [hunk 1]
```diff
@@ -798,6 +798,9 @@ int prompt_for_number(int *mouse_used)
     cmdline_row = save_cmdline_row;
   }
   State = save_State;
+
+  ui_call_msg_clear();
```
Comment (bfredl) [inline src/nvim/misc1.c:802]: too early, doing something equivalent to `echo inputlist(["aa","bb"]) + inputlist(["cc", "dd"])` should not hide the message after first input. If the issue is specific to spell better fix it in `spell_suggest()`

Comment (glacambre) [reply src/nvim/misc1.c]: As far as I could tell, `echo inputlist(["aa", "bb"]) + inputlist(["cc", "dd"])` had the same behavior with and without my patch.

I've changed it and forced-pushed your suggestion anyway :).

File: src/nvim/spell.c [hunk 2]
```diff
@@ -2889,6 +2890,9 @@ void spell_suggest(int count)
     msg_col = 0;
     // Ask for choice.
     selected = prompt_for_number(&mouse_used);
+
+    ui_call_msg_clear();
```
Comment (bfredl) [inline src/nvim/spell.c:2894]: one more thing, we want `if ui_has(kUiMessages)` around it :)

Comment (glacambre) [reply src/nvim/spell.c:2894]: Should I also fix the linting issues around my patch or leave it alone?

Comment (bfredl) [reply src/nvim/spell.c:2894]: yes, if it is not too much :)

File: test/functional/ui/messages_spec.lua [hunk 3]
```diff
@@ -795,6 +795,51 @@ describe('ui/ext_messages', function()
...
+    screen:expect{grid=[[
+      {r:helllo}                   |
+      {b:~                        }|
+      {b:~                        }|
+      {b:~                        }|
+      {b:^~                        }|
+    ]], messages={
+      {content = { { 'Change "helllo" to:\n 1 "Hello"\n 2 "Hallo"\n 3 "Helli"\nType number and <Enter> or click with mouse (empty cancels): ' } }, kind = ""}
+    }, attr_ids = {
+      r = {special = Screen.colors.Red, undercurl = true},
```
Comment (bfredl) [inline test/functional/ui/messages_spec.lua:813]: add this to `screen:set_default_attr_ids{}` in `before_each` instead.

File: test/functional/ui/messages_spec.lua [hunk 4]
```diff
@@ -795,6 +795,51 @@ describe('ui/ext_messages', function()
...
+    }}
+
+    feed('<cr>')
+    screen:expect{grid=[[
+      ^Hello                    |
+      {1:~                        }|
+      {1:~                        }|
+      {1:~                        }|
+      {1:~                        }|
+    ]], messages={}}
```
Comment (bfredl) [inline test/functional/ui/messages_spec.lua:839]: empty messages key can be removed (we always assert against unexpected ext_ taste)

**Comment by glacambre** [comment]
Thank you for taking the time to review this and all the patient advice! :)

**Pull request merged**